src/main/java/teammates/ui/controller/StudentHomePageAction.java

```diff
@@ -26,9 +26,14 @@ public ActionResult execute() throws EntityDoesNotExistException {
         String recentlyJoinedCourseId = getRequestParamValue(Const.ParamsNames.CHECK_PERSISTENCE_COURSE);        
         
         try {
-            data.courses = logic.getCourseDetailsListForStudent(account.googleId);
-    
-            data.sessionSubmissionStatusMap = generateFeedbackSessionSubmissionStatusMap(data.courses, account.googleId);
+            Map<String, StudentAttributes> courseToAttributesMap = new HashMap<String, StudentAttributes>();
+            data.courses = logic.getCourseDetailsListForStudent(account.googleId, courseToAttributesMap);
+            /*
+             * courseToAttributesMap is used to keep the query results in the form of
+             * course ID -> StudentAttributes mapping so that they can be re-used.
+             */
+            data.sessionSubmissionStatusMap = generateFeedbackSessionSubmissionStatusMap(data.courses, account.googleId,
+                                                                                         courseToAttributesMap);
```

Comment:
`courseToAttributes` is ambiguous. What `Attributes`? May be use `courseStudentMap`?